bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -141,9 +152,25 @@ public Builder setStatsLogger(StatsLogger statsLogger) {
             return this;
         }
 
+
+        public Builder dnsResolver(DNSToSwitchMapping dnsResolver) {
+            this.dnsResolver = dnsResolver;
+            return this;
+        }
+
+        public Builder requestTimer(HashedWheelTimer requestTimer) {
```

Comment:
it is necessary to expose requestTimer to clients ? 
for this version it is used only internally by the new policy, exposing it to clients will make BookKeeper support this property for next releases.
I wonder if there could be a way to such kind of shared resources to be used only for specific cases